src/internal_events/http.rs

```diff
@@ -1,27 +1,65 @@
-// ## skip check-events ##
-
 use metrics::counter;
 use std::error::Error;
 use vector_core::internal_event::InternalEvent;
 
 #[derive(Debug)]
-pub struct HttpEventsReceived {
-    pub events_count: usize,
+pub struct HttpBytesReceived<'a> {
+    pub byte_size: usize,
+    pub http_path: &'a str,
+    pub protocol: &'static str,
+}
+
+impl InternalEvent for HttpBytesReceived<'_> {
+    fn emit_logs(&self) {
+        trace!(
+            message = "Received bytes.",
+            byte_size = %self.byte_size,
+            http_path = %self.http_path,
+            protocol = %self.protocol
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "component_received_bytes_total", self.byte_size as u64,
+            "http_path" => self.http_path.to_string(),
+            "protocol" => self.protocol,
+        );
+    }
+}
+
+#[derive(Debug)]
+pub struct HttpEventsReceived<'a> {
+    pub count: usize,
     pub byte_size: usize,
+    pub http_path: &'a str,
+    pub protocol: &'static str,
 }
 
-impl InternalEvent for HttpEventsReceived {
+impl InternalEvent for HttpEventsReceived<'_> {
     fn emit_logs(&self) {
         trace!(
             message = "Received events.",
-            events_count = %self.events_count,
+            count = %self.count,
             byte_size = %self.byte_size,
```

Comment:
Just curious, why is it important to have both total byte size received and event byte size?

Reply:
I'm not entirely sure, but that's the way the spec is written. It's a bit of a before and after conversion value, as the one describes the raw bytes on the wire, which may be compressed or expanded by escapes, and the other describes the internal representation, for comparison with other such metrics.